src/libLookup/Lookup.cpp

```diff
@@ -1892,9 +1944,9 @@ bool Lookup::ProcessSetStateFromSeed(const vector<unsigned char>& message,
       }
       m_currDSExpired = false;
     }
-  } else if (m_syncType == SyncType::LOOKUP_SYNC) {
-    // rsync the txbodies here
-    if (RsyncTxBodies() && !m_currDSExpired) {
+  } else if (m_syncType == SyncType::LOOKUP_SYNC ||
+             m_syncType == SyncType::NEW_LOOKUP_SYNC) {
+    if (!m_currDSExpired) {
```

Comment:
I see that whenever state is recvd it will finish joining/ shouldnt there be a mechanism to check if all the others are also fetched

Reply:
yeah. fixed. State processing is now done only after TxnBlocks are committed. Also, we make sure DS info is processed before marking finish of new join.